hpx/parallel/util/scan_partitioner.hpp

```diff
@@ -0,0 +1,269 @@
+//  Copyright (c) 2007-2015 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_PARALLEL_UTIL_SCAN_PARTITIONER_DEC_30_2014_0227PM)
+#define HPX_PARALLEL_UTIL_SCAN_PARTITIONER_DEC_30_2014_0227PM
+
+#include <hpx/hpx_fwd.hpp>
+#include <hpx/async.hpp>
+#include <hpx/exception_list.hpp>
+#include <hpx/lcos/wait_all.hpp>
+#include <hpx/lcos/local/dataflow.hpp>
+#include <hpx/util/bind.hpp>
+#include <hpx/util/decay.hpp>
+
+#include <hpx/parallel/execution_policy.hpp>
+#include <hpx/parallel/util/detail/chunk_size.hpp>
+#include <hpx/parallel/util/detail/handle_local_exceptions.hpp>
+#include <hpx/parallel/traits/extract_partitioner.hpp>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace hpx { namespace parallel { namespace util
+{
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        ///////////////////////////////////////////////////////////////////////
+        // The static partitioner simply spawns one chunk of iterations for
+        // each available core.
+        template <typename ExPolicy, typename R, typename Result = void>
+        struct static_scan_partitioner
+        {
+            template <typename FwdIter, typename T,
+                typename F1, typename F2, typename F3>
+            static R call(ExPolicy const& policy, FwdIter first,
+                std::size_t count_, T && init, F1 && f1, F2 && f2, F3 && f3,
+                std::size_t chunk_size)
+            {
+                std::vector<hpx::shared_future<Result> > workitems;
+                std::vector<std::size_t> chunk_sizes;
+
+                std::list<boost::exception_ptr> errors;
+
+                try {
+                    // pre-initialize first intermediate result
+                    workitems.push_back(make_ready_future(std::forward<T>(init)));
+
+                    // estimate a chunk size based on number of cores used
+                    std::size_t count = count_;
+                    chunk_size = get_static_chunk_size(policy, workitems, f1,
+                        first, count, chunk_size);
+
+                    HPX_ASSERT(workitems.size() == 2);
+
+                    // schedule every chunk on a separate thread
+                    workitems.reserve(count_ / chunk_size + 2);
+                    chunk_sizes.reserve(workitems.capacity());
+
+                    chunk_sizes.push_back(count_ - count);
+
+                    // pre-initialize second intermediate result
+                    workitems[1] = lcos::local::dataflow(hpx::launch::sync,
+                        f2, workitems[0], workitems[1]);
```

Comment:
I can see the assert, but think it is firing. This changeset doesn't change workitems in get_static_chunk_size, as far as i can see it. Where is the test verifying that it doesn't fire?